Add deprecation message for Enable/DisablePersistentKernelCache

ruizhangTT opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/32208

### Problem description
According to API stability policy, the APIs will be marked as deprecated and deleted later.

### What's changed
Added deprecation message.

### Checklist
No functional change; a passing build should suffice.

ruizhangTT committed:
Add deprecation message for EnablePersistentKernelCache

akerteszTT commented on tt_metal/api/tt-metalium/persistent_kernel_cache.hpp:
```diff
@@ -14,13 +12,15 @@ namespace tt::tt_metal::detail {
  *
  * Return value: void
  */
+[[deprecated("Persistent kernel cache is no longer needed as the JIT build system now supports caching by default.")]]
```
Could you please also include in your message the text. "This API will be removed after <date>"?
Under the Moreh agreement we hashed out, <date> has to be at least 4 weeks out :( I realize that PRs can sometimes stay open a while, so this is a bit of a pain, but can always make one last string change before a commit.

Though not critical for your PR today, would be good to role model the new methodology (for the dev team and for Moreh).

akerteszTT reviewed (approved):
Thanks for adapting your PR to the new Moreh deprecation scheme! I realize it's a pain, so much appreciated :)

Please tweak the deprecation string before merging.

ruizhangTT committed:
Enhance deprecation notice for kernel cache API

Updated deprecation message for persistent kernel cache functions to include removal date.
